### PR TITLE
Add revapi as option during build

### PR DESCRIPTION
**Pull request by maxandersen**
This is related to #10453 but *just* includes the minimal setup to run revapi (no modules gets it enabled; that will be future PR's) 

Initial setup of revapi to get API change reports.
Still ways to go in tuning it; but would like to have it included in master
now to make coming PR's simpler to apply/manage.

This adds a profile to run api check - it is not currently activated by default
but should be in the future when we have the defaults tuned in to not generate
unwarranted noise.

Also `revapi.skip` is default true, meaning even when the profile is active no api check
will be done on a module unless it defines `revapi.skip=true` in its properties.

Thus for now to even have it run you need enable the api-check profile *and* the property.
i.e. `mvn -Papi-check -Drevapi.skip=false`

If you want to run it for everything (ignoring revapi.skip do the following::
`mvn -Papi-check -Drevapi.skip=false -DskipTests verify`

If you want to see it detect changes in dependencies that quarkus exposes/uses add `-Drevapi.checkdeps=true`

When run you should get a `target/revapi-results/` in every module that are included in the check.

That folder will have a `.json` and `.adoc` file capturing detected API changes/issues.

If you want to tweak the format of the `.adoc` file you can change that in revapi/**/*.ftl
and do a `mvn install` to have changes picked up.

### CODE REVIEW

**Comment by geoand** [comment]
I don't think I can meaningfully review this before I get back from PTO.

But as soon as I do, it will be one of the first things I do

**Comment by geoand** [comment]
I just saw that this just configures revapi and doesn't enable it in any way.

I would still prefer if someone else pulled the trigger on it as I won't really be around for follow-ups and subsequent discussions.

**Comment by maxandersen** [comment]
> I don't think I can meaningfully review this before I get back from PTO.
> 
> But as soon as I do, it will be one of the first things I do

understood - there isn't much to it though; should have zero effect on build both locally and in CI but lets us try out revapi configs in tiny PR updates.

File: build-parent/pom.xml [hunk 1]
```diff
@@ -523,9 +548,7 @@
             <id>format</id>
             <activation>
                 <activeByDefault>true</activeByDefault>
-                <property>
-                    <name>!no-format</name>
-                </property>
```
Comment (gsmet) [inline build-parent/pom.xml:528]: I don't understand this change.

**Review by gsmet — CHANGES_REQUESTED** [review]
Want to clarify something before thinking about merging.

File: build-parent/pom.xml (hunk 1 above)
Comment (maxandersen) [reply build-parent/pom.xml]: neither do it ;/

wrong profile it was done to - pushed a fix.

**Comment by gsmet** [comment]
I made some changes to disable the profile by default: it now needs to be enabled explicitly.

**Review by gsmet — APPROVED** [review]

**Comment by maxandersen** [comment]
thanks @gsmet - merged with ci passing and i'll updated #10453 to just have the new delta(s)

**Pull request merged**